### DISCUSSION TITLE
How to fetch on server before rendering?

### DISCUSSION THREAD

**Discussion by Torsten85** [solidjs/solid-start#1421] [Q&A] [answered]
Hello,
I've read the data loading part of the docs, but that solution seems to load the data after the inital rendering so it won't work without javascript and the data isn't in the initial html. Is there a way to hold rendering on the server until all data is fetched so the initial render is "complete" without any loading states, white screens? 

Found the exact same question here (https://github.com/solidjs/solid-start/discussions/1107) but that solution isn't valid anymore with RC 1.

**Comment by tayllana** [comment]


**Reply by Torsten85** [reply]
this AI nonsense has to stop

**Comment by lxsmnsyc** [comment, marked as answer]
Adding `deferStream` to a resource should allow the resource to block the SSR before sending the initial HTML.

```js
const [data] = createResource(fetchData, {
  deferStream: true,
});
```

**Reply by Torsten85** [reply]
Thanks! 

Just looked into the source, and the "deferStream" option also exists for "createAsync" even though its missing in the docs.
